pkg/network/event_common.go

```diff
@@ -118,26 +118,59 @@ type BufferedData struct {
 type Connections struct {
 	BufferedData
 	DNS                         map[util.Address][]string
-	ConnTelemetry               *ConnectionsTelemetry
+	ConnTelemetry               map[ConnTelemetryType]int64
 	CompilationTelemetryByAsset map[string]RuntimeCompilationTelemetry
 	HTTP                        map[http.Key]http.RequestStats
 	DNSStats                    dns.StatsByKeyByNameByType
 }
 
-// ConnectionsTelemetry stores telemetry from the system probe related to connections collection
-type ConnectionsTelemetry struct {
-	MonotonicKprobesTriggered          int64
-	MonotonicKprobesMissed             int64
-	MonotonicConntrackRegisters        int64
-	MonotonicConntrackRegistersDropped int64
-	MonotonicDNSPacketsProcessed       int64
-	MonotonicConnsClosed               int64
-	ConnsBpfMapSize                    int64
-	MonotonicUDPSendsProcessed         int64
-	MonotonicUDPSendsMissed            int64
-	ConntrackSamplingPercent           int64
-	DNSStatsDropped                    int64
-}
+// ConnTelemetryType enumerates the connection telemetry gathered by the system-probe
+// The string name of each telemetry type is the metric name which will be emitted
+type ConnTelemetryType string
+
+//revive:disable:exported
+const (
+	MonotonicKprobesTriggered          ConnTelemetryType = "kprobes_triggered"
+	MonotonicKprobesMissed                               = "kprobes_missed"
+	MonotonicConnsClosed                                 = "conntrack_registers"
+	MonotonicConntrackRegisters                          = "conntrack_registers_dropped"
+	MonotonicConntrackRegistersDropped                   = "dns_packets_processed"
+	MonotonicDNSPacketsProcessed                         = "conns_closed"
+	MonotonicUDPSendsProcessed                           = "udp_sends_processed"
+	MonotonicUDPSendsMissed                              = "udp_sends_missed"
+	DNSStatsDropped                                      = "dns_packets_dropped"
+	ConnsBpfMapSize                                      = "conns_bpf_map_size"
+	ConntrackSamplingPercent                             = "conntrack_sampling_percent"
+	NPMDriverFlowsMissedMaxExceeded                      = "dns_stats_dropped"
+	MonotonicDNSPacketsDropped                           = "driver_flows_missed_max_exceeded"
```

Comment:
It seems a bunch of the strings don't match the names?

Reply:
🧐  that is... definitely not right. Fixed!